include/fiction/algorithms/simulation/sidb/sidb_simulation_result.hpp

```diff
@@ -59,6 +63,59 @@ struct sidb_simulation_result
      * The key of the map is the name of the parameter, the element is the value of the parameter.
      */
     std::unordered_map<std::string, std::any> additional_simulation_parameters{};
+    /**
+     * This function calculates the ground state charge distributions from the provided simulation results.
+     * The ground state charge distributions are those with energy closest to the minimum energy found in the simulation
+     * results.
+     *
+     * @note When degenerate states exist, there are multiple ground states with the same energy.
+     *
+     * @return A vector of charge distributions with the minimal energy.
+     */
+    [[nodiscard]] std::vector<charge_distribution_surface<Lyt>> groundstates() const noexcept
+    {
+        std::vector<charge_distribution_surface<Lyt>> groundstate_charge_distributions{};
+        std::set<uint64_t>                            charge_indices{};
+
+        // in case, the charge indices are not updated.
+        auto charge_configurations_copy = charge_distributions;
+
+        // Find all unique charge indices. This is done because simulation results can have multiple identical charge
+        // distributions.
+        for (auto& cds : charge_configurations_copy)
+        {
+            cds.charge_distribution_to_index();
+            charge_indices.insert(cds.get_charge_index_and_base().first);
+        }
+
+        // Find the minimum energy
+        double min_energy = std::numeric_limits<double>::infinity();
+        if (!charge_configurations_copy.empty())
+        {
+            min_energy = std::min_element(charge_configurations_copy.begin(), charge_configurations_copy.end(),
+                                          [](const auto& lhs, const auto& rhs)
+                                          { return lhs.get_system_energy() < rhs.get_system_energy(); })
+                             ->get_system_energy();
+        }
```

Comment:
Don't we have an existing function for minimum energy?

Reply:
You are absolutely right! This is still old code. Thanks!